specification/iotcentral/data-plane/Microsoft.IoTCentral/stable/2022-07-31/iotcentral.json

```diff
@@ -0,0 +1,6113 @@
+      "name": "componentName",
+      "description": "Name of the device component.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "DeviceGroupId": {
+      "in": "path",
+      "name": "deviceGroupId",
+      "description": "Unique ID for the device group.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "maxLength": 255,
+      "pattern": "^[a-zA-Z0-9_-]*$",
+      "required": true
+    },
+    "DeviceId": {
+      "in": "path",
+      "name": "deviceId",
+      "description": "Unique ID of the device.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "DeviceTemplateId": {
+      "in": "path",
+      "name": "deviceTemplateId",
+      "description": "Unique [Digital Twin Model Identifier](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#digital-twin-model-identifier) of the device template.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "EnrollmentGroupId": {
+      "in": "path",
+      "name": "enrollmentGroupId",
+      "description": "Unique ID of the enrollment group.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "maxLength": 255,
+      "pattern": "^[a-zA-Z0-9-]*$",
+      "required": true
+    },
+    "Entry": {
+      "in": "path",
+      "name": "entry",
+      "description": "Entry of certificate only support primary and secondary.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "enum": [
+        "primary",
+        "secondary"
+      ],
+      "x-ms-enum": {
+        "name": "CertificateEntry",
+        "modelAsString": false,
+        "values": [
+          {
+            "value": "primary",
+            "name": "Primary",
+            "description": "The entry of primary certificate"
+          },
+          {
+            "value": "secondary",
+            "name": "Secondary",
+            "description": "The entry of secondary certificate"
+          }
+        ]
+      },
+      "required": true
+    },
+    "Filter": {
+      "in": "query",
+      "name": "$filter",
+      "description": "An expression on the resource type that selects the resources to be returned.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": false
+    },
+    "JobId": {
+      "in": "path",
+      "name": "jobId",
+      "description": "Unique ID of the job.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "maxLength": 255,
+      "pattern": "^[a-zA-Z0-9_-]*$",
+      "required": true
+    },
+    "ModuleName": {
+      "in": "path",
+      "name": "moduleName",
+      "description": "Name of the device module.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "OrderBy": {
+      "in": "query",
+      "name": "$orderby",
```

Comment:
Don't use "$" prefix on parameter names.

Reply:
Removed $filter, $top, $orderby from July GA, will revise this and get preview on Sep.